lib/internal/blob.js

```diff
@@ -352,13 +356,15 @@ class Blob {
               pending.reject(error);
               return;
             }
-            if (buffer !== undefined) {
+            // ReadableByteStreamController.enqueue errors if we submit a 0-length
+            // buffer. We need to check for that here.
+            if (buffer !== undefined && buffer.byteLength !== 0) {
               c.enqueue(new Uint8Array(buffer));
             }
             // We keep reading until we either reach EOS, some error, or we
             // hit the flow rate of the stream (c.desiredSize).
             queueMicrotask(() => {
-              if (c.desiredSize <= 0) {
+              if (c.desiredSize < 0) {
```

Comment:
Would draw attention of reviewers to this specific update it seems when we turn this into a ReadableByteStream the high water mark for bytestreams is 0 and for normal default streams is 1 hence we have to remove the equality here 
see:
https://github.com/nodejs/node/blob/2ccfb23b7fc8e938a87d52949bf24e9720f846b1/lib/internal/webstreams/readablestream.js#L282-L298

also each byte is considered to be an element in the case of byte streams and hence the `desiredSize` can go below 0, I nonetheless asserted that the memory isnt overflowing in the existing tests but do take a look at them too if they look correct!

Thank You!